### ISSUE TITLE
Playbook schema should allow lists of lists of strings for vars_files

### ISSUE THREAD

**Issue by vaygr** [ansible/ansible-lint#3558]
##### Summary

As per https://github.com/ansible/ansible-lint/issues/2474#issuecomment-1262314661 and [Conditionals doc](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_conditionals.html#selecting-variables-files-based-on-facts) `vars_files` must allow lists of lists of strings like so:

```yaml
  vars_files:
    - ["vars/{{ ansible_os_family | lower }}.yml", "vars/default.yml"]
```

to achieve 'first found' logic.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint 6.15.1.dev45 using ansible 2.15.0
```

- ansible installation method: official Manjaro package
- ansible-lint installation method: official Manjaro package

##### STEPS TO REPRODUCE

```console (paste below)
---
- name: Configure basesystem
  hosts: all
  become: true
  vars_files:
    - ["vars/{{ ansible_os_family | lower }}.yml", "vars/default.yml"]
```

##### Desired Behavior

No errors reported by ansible-lint.

##### Actual Behavior

```paste below
WARNING  Listing 1 violation(s) that are fatal
schema[playbook]: ['vars/{{ ansible_os_family | lower }}.yml', 'vars/default.yml'] is not of type 'string'
main.yml:1  Returned errors will not include exact line numbers, but they will mention
the schema name being used as a tag, like ``schema[playbook]``,
``schema[tasks]``.

This rule is not skippable and stops further processing of the file.

If incorrect schema was picked, you might want to either:

* move the file to standard location, so its file is detected correctly.
* use ``kinds:`` option in linter config to help it pick correct file type.


Read documentation for instructions on how to ignore specific rule violations.

               Rule Violation Summary
 count tag              profile rule associated tags
     1 schema[playbook] basic   core

Failed after min profile: 1 failure(s), 0 warning(s) on 108 files.
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve
